src/lib/process/index.ts

```diff
@@ -17,14 +19,14 @@ export const createProcess = async (
   dryRun: boolean = false,
   options: Polkadot.Options = defaultOptions
 ): Promise<Process.Result> => {
-  const restrictions: Restrictions.Restrictions = mapRestrictions(rawRestrictions)
+  const restrictions: ChainRestrictions = mapRestrictions(rawRestrictions)
   const processId = utf8ToHex(name, Constants.PROCESS_ID_LENGTH)
 
   const polkadot: Polkadot.Polkadot = await createNodeApi(options)
   const currentVersion = await getVersion(polkadot, processId)
 
   if (version !== currentVersion + 1) {
-    throw new Error(`Version: ${version} must be one higher than current version: ${currentVersion}`)
+    throw new VersionError(`Version: ${version} must be one higher than current version: ${currentVersion}`)
```

Comment:
Nice

Reply:
would maybe cut the size of message as some might be possible to abstract e.g. could take args **version**, **currentVersion**, but anyway nice for doing custom errors.

Reply:
there's another type of `VersionError` with a different format, although maybe could argue that it's a different kind of error `VersionError('Version: ${processVersion} does not exist for name: ${name}')`